### ISSUE TITLE
Clean up dealing dialog details

### ISSUE THREAD

**Issue by joshuaauerbachwatson** [joshuaauerbachwatson/anyCards#61]
Split from issue #60.

1.  The "Deal" button has the wrong enablement initially.  Toggling the hand area off and on seems to fix this.
2.  The Deal dialog should show an error when the number of cards required by the deal exceeds the size of the deck.   This was omitted from the rewrite.
3. The Deal dialog should dismiss itself when the deal is complete.   Now, it requires touching outside the dialog to make it go away.
4. Reset button does not work.
5. Target boxes are placed a bit too low: the legend overlaps the hand area marker.   When there is no hand area, they are in the view but the view is not constrained by the safe area at the bottom.
6. Probably dealing to "hands" should not be offered if there is no hands area because hands are designed to be "taken" by individual players.

**Comment by joshuaauerbachwatson** [comment]
> Target boxes are placed a bit too low: the legend overlaps the hand area marker. When there is no hand area, they are in the view but the view is not constrained by the safe area at the bottom.

I think this may be due to an old bug still present in `PlayingSurface`.  Testing that hypothesis.

> The "Deal" button has the wrong enablement initially. Toggling the hand area off and on seems to fix this.

Ending the game and starting again also fixes it.  It doesn't recur.  It seems to be an artifact of a completely fresh app instance.

**Comment by joshuaauerbachwatson** [comment]
Items (1) and (5) are now resolved.

**Comment by joshuaauerbachwatson** [comment]
Item (1) was not really resolved; I had omitted the check for the presence of the deck.   Restoring the check (while still ensuring that the view is initialized) the button is improperly disabled, so still have to investigate that.

Meanwhile, items (2) and (3) are now resolved.

**Comment by joshuaauerbachwatson** [comment]
Concerning item (1): it was actually correct to omit the test for `deck == nil` and I have returned to testing only the `canDeal` flag.   For the moment, there is an invariant that a playing surface must always have a deck.  We _could_ eventually change the model so that a "new" playing surface never has a deck and decks must always be added if desired.   But, I was getting myself in trouble by trying to have it both ways.

Item (6) is now addressed.

This leaves only item (4) (the reset button).   The action for the reset button was always TODO so I now need to implement it.
